pkg/analysis/registry.go

```diff
@@ -69,36 +58,31 @@ type Registry interface {
 }
 
 type registry struct {
-	initializers []AnalyzerInitializer
+	lock         sync.RWMutex
+	initializers []initializer.AnalyzerInitializer
 }
```

Comment:
Asking this because we've added a mutex which makes this thread safe - should we attempt to handle the case where we've performed initialization and then a new initializer gets registered?

Would we consider that a valid and supported use case or should this block registration once initialization has been run?